src/main/java/com/bloxbean/cardano/client/backend/api/helper/impl/UtxoTransactionBuilderImpl.java

```diff
@@ -33,6 +37,14 @@
 @Slf4j
 public class UtxoTransactionBuilderImpl implements UtxoTransactionBuilder {
 
+    @Data
+    @AllArgsConstructor
+    private class PaymentTransactionGroupingKey {
+        private Account sender;
+        private String receiver;
+        private String datumHash;
+    }
+
```

Comment:
This `case class` identifies the transaction grouping rationale. Basically in order to collate transactions in just one input:
1. sender
2. receiver
3. datum (hash)
need to be the same.